src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -276,4 +252,15 @@ ApplyLedgerChainWork::onRun()
         return State::WORK_FAILURE;
     }
 }
+
+void
+ApplyCheckpointWork::onSuccess()
+{
+    // Application completed successfully, safe to delete files
```

Comment:
How are those files deleted on error?

Reply:
Catchup fails, and the whole tmp dir is removed via RAII (this is how these files are currently handled on both error and success)